### ISSUE TITLE
no-useless-constructor: class declaration with constructor throws exception

### ISSUE THREAD

**Issue by JamesHenry** [typescript-eslint/typescript-eslint#15]
This issue was initially reported here: https://github.com/eslint/typescript-eslint-parser/issues/485

---

**What version of TypeScript are you using?**
2.9.1

**What version of `typescript-eslint-parser` are you using?**
16.0.0

**What code were you trying to parse?**

```ts
// foo.ts
declare class Foo {
  constructor();
}
```

or

``` ts
// foo.d.ts
class Foo {
  constructor();
}
```

**What did you expect to happen?**
No exception.

**What happened?**
```
TypeError: Cannot read property 'body' of null
    at checkForConstructor (/Users/OliverJAsh/Development/unsplash-web/node_modules/eslint/lib/rules/no-useless-constructor.js:167:41)
    at listeners.(anonymous function).forEach.listener (/Users/OliverJAsh/Development/unsplash-web/node_modules/eslint/lib/util/safe-emitter.js:47:58)
    at Array.forEach (native)
    at Object.emit (/Users/OliverJAsh/Development/unsplash-web/node_modules/eslint/lib/util/safe-emitter.js:47:38)
    at NodeEventGenerator.applySelector (/Users/OliverJAsh/Development/unsplash-web/node_modules/eslint/lib/util/node-event-generator.js:251:26)
    at NodeEventGenerator.applySelectors (/Users/OliverJAsh/Development/unsplash-web/node_modules/eslint/lib/util/node-event-generator.js:280:22)
    at NodeEventGenerator.enterNode (/Users/OliverJAsh/Development/unsplash-web/node_modules/eslint/lib/util/node-event-generator.js:294:14)
    at CodePathAnalyzer.enterNode (/Users/OliverJAsh/Development/unsplash-web/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:608:23)
    at Traverser.enter [as _enter] (/Users/OliverJAsh/Development/unsplash-web/node_modules/eslint/lib/linter.js:865:28)
    at Traverser._traverse (/Users/OliverJAsh/Development/unsplash-web/node_modules/eslint/lib/util/traverser.js:132:14)
```

Possibly related to https://github.com/eslint/typescript-eslint-parser/issues/384?

**Comment by foisonocean** [comment]
Is there any solutions for that? This make my CI failed.

**Comment by platinumazure** [comment]
@foisonocean Looks like this should be fixed in 1.1.0. Are you using that version? If so and you still see this issue, please open a new issue. Thanks!

**Comment by jacobrask** [comment]
Has the fix for this been released?

**Comment by bradzacher** [comment]
hey @jacobrask - please have a look at the release logs instead of commenting on old threads.
The last message in this issue is that it was closed by pr #167.

The release logs list #167 released as part of 1.2.0
https://github.com/typescript-eslint/typescript-eslint/releases
